### PR TITLE
SUMO-191578, made resolution window a optional field and added test c…

### PR DESCRIPTION
**Pull request by rmamilla-sumo**
…ases

### CODE REVIEW

File: sumologic/resource_sumologic_monitors_library_monitor.go [hunk 1]
```diff
@@ -386,9 +386,13 @@ var logsStaticTriggerConditionSchema = map[string]*schema.Schema{
 			"threshold_type": &thresholdTypeSchema,
 		}),
 		"resolution": nested(false, schemaMap{
-			"threshold":         &thresholdSchema,
-			"threshold_type":    &thresholdTypeSchema,
-			"resolution_window": &timeRangeSchema,
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+			"resolution_window": {
```
Comment (tarunk2) [inline sumologic/resource_sumologic_monitors_library_monitor.go:391]: DRY - We can define a `var` for it and re-use at all these places.

**Review by tarunk2 — APPROVED** [review]
One comment. LGTM once that is addressed.

File: sumologic/resource_sumologic_monitors_library_monitor.go (hunk 1 above)
Comment (rmamilla-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go:391]: For some reason it didn't reference the variable timeRangeSchema so went this approach. Will explore more

Comment (rmamilla-sumo) [reply sumologic/resource_sumologic_monitors_library_monitor.go]: Done.

**Comment by rmamilla-sumo** [comment]
Not able to write a negative test case to check if -ve value for resolution window fails but did an external test by running terraform validate and it is throwing an error

 Error: invalid value for triggers.1.resolution_window (Resolution window must be in the format '\d+[smhd]'. Examples: 0m, 15m, 1d, etc.)
│ 
│   with sumologic_monitor.tf_logs_monitor_1,
│   on main.tf line 32, in resource "sumologic_monitor" "tf_logs_monitor_1":
│   32: resource "sumologic_monitor" "tf_logs_monitor_1" {



For Input
triggers  {
		threshold_type = "LessThanOrEqual"
		threshold = 40.0
		time_range = "-30m"
		occurrence_type = "ResultCount"
		trigger_source = "AllResults"
		trigger_type = "ResolvedCritical"
		detection_method = "StaticCondition"
		resolution_window = "-15m"
	}

**Review by tarunk2 — APPROVED** [review]

**Pull request merged**